### PR TITLE
EVG-992: create enclosing directories if needed before s3.get and json.get commands

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: command/taskdata_get.go [hunk 1]
```diff
@@ -51,6 +51,11 @@ func (c *taskDataGet) Execute(ctx context.Context,
 	if c.File != "" && !filepath.IsAbs(c.File) {
 		c.File = filepath.Join(conf.WorkDir, c.File)
 	}
+
+	if err = createEnclosingDirectoryIfNeeded(c.File); err != nil {
```
Comment (bsamek) [inline command/taskdata_get.go]: This differs from the s3 case. Here we create the directory regardless of whether the path is absolute. Should we only create the directory if the path is relative?

Comment (tychoish) [reply command/taskdata_get.go]: I've clarified this in the next commit

File: command/taskdata_get.go [hunk 2]
```diff
@@ -48,7 +48,7 @@ func (c *taskDataGet) Execute(ctx context.Context,
 		return err
 	}
 
-	if c.File != "" && !filepath.IsAbs(c.File) {
```
Comment (bsamek) [inline command/taskdata_get.go]: Why remove this check?

Comment (tychoish) [reply command/taskdata_get.go]: we do it on line 30

**Pull request merged**